routing/missioncontrol.go

```diff
@@ -299,6 +258,30 @@ func (m *MissionControl) getEdgeProbabilityForNode(nodeHistory *nodeHistory,
 	return probability
 }
 
+// getPairProbability estimates the probability of successfully
+// traversing from fromNode to toNode based on historical payment outcomes.
+func (m *MissionControl) getPairProbability(fromNode,
+	toNode route.Vertex, amt lnwire.MilliSatoshi) float64 {
+
+	// Start by getting the last node level failure. If there is none,
+	// lastFail will be zero.
+	lastFail := m.lastNodeFailure[fromNode]
+
+	// Retrieve the last pair outcome.
+	pair := NewDirectedNodePair(fromNode, toNode)
+	lastPairResult, ok := m.lastPairFailure[pair]
+
+	// If there is none or it happened before the last node level failure,
+	// the node level failure is the most recent and thus returned.
+	if ok && lastPairResult.timestamp.After(lastFail) {
```

Comment:
I think the prior comment from `getEdgeProbabilityForNode` should be ported here. Specifically the portion explaining the min penalize amount and balance related errors.

Reply:
Yes, those got lost. Added them back in again. Good point